### ISSUE TITLE
Severe JVM Render Crash (According to someone in the Modded Minecraft Discord server)

### ISSUE THREAD

**Issue by Musikid-Official** [Nettakrim/Signed-Paintings#7]
# Severe JVM Rendering Error (apparently, i don't have the braincells to understand java)

I pasted an imgur link ending with `.jpg` and it crashed my game. Here's the snippet of my log in which this happened (full log didn't fit):

```[20:22:41] [Game thread/INFO]: Loaded 186 advancements
FATAL ERROR in native method: Thread[ForkJoinPool.commonPool-worker-2,10,main]: No context is current or a function that is not available in the current context was called. The JVM will abort execution.
    at org.lwjgl.opengl.GL11C.nglGenTextures(Native Method)
    - parking to wait for  <0x0000000726e4c820> (a java.util.concurrent.ForkJoinPool)
    at org.lwjgl.opengl.GL11C.glGenTextures(GL11C.java:667)
    at org.lwjgl.opengl.GL11.glGenTextures(GL11.java:2439)
    at com.mojang.blaze3d.platform.GlStateManager._genTexture(GlStateManager.java:562)
    at com.mojang.blaze3d.platform.TextureUtil.generateTextureId(TextureUtil.java:45)
    at net.minecraft.class_1044.method_4624(class_1044.java:41)
    at net.minecraft.class_1043.<init>(class_1043.java:28)
    at com.nettakrim.signed_paintings.util.ImageManager.saveBufferedImageAsIdentifier(ImageManager.java:183)
    at com.nettakrim.signed_paintings.util.ImageManager.onImageLoad(ImageManager.java:156)
    at com.nettakrim.signed_paintings.util.ImageManager.lambda$registerImage$0(ImageManager.java:144)
    at com.nettakrim.signed_paintings.util.ImageManager$$Lambda$15745/0x00000008028e1498.apply(Unknown Source)
    at java.util.concurrent.CompletableFuture.uniHandle(java.base@17.0.7/CompletableFuture.java:934)
    at java.util.concurrent.CompletableFuture$UniHandle.tryFire(java.base@17.0.7/CompletableFuture.java:911)
    at java.util.concurrent.CompletableFuture$Completion.exec(java.base@17.0.7/CompletableFuture.java:483)
    at java.util.concurrent.ForkJoinTask.doExec(java.base@17.0.7/ForkJoinTask.java:373)
    at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(java.base@17.0.7/ForkJoinPool.java:1182)
    at java.util.concurrent.ForkJoinPool.scan(java.base@17.0.7/ForkJoinPool.java:1655)
    at java.util.concurrent.ForkJoinPool.runWorker(java.base@17.0.7/ForkJoinPool.java:1622)
    at java.util.concurrent.ForkJoinWorkerThread.run(java.base@17.0.7/ForkJoinWorkerThread.java:165)
Process exited with code 1.```

**Comment by Musikid-Official** [comment]
Updated my graphics drivers, didn't fix the issue.

Specs, in case it matters:

OS - Win11
GPU - RTX 3060 Ti
CPU - Ryzen 7 3700X
RAM - 64GB DDR4 @ 2133MHz

**Comment by Nettakrim** [comment]
do you have the specific link / do other jpgs crash? (im pretty sure jpgs have worked for me in the past)

this is definately a spooky looking crash report that might be unlikely to be able to fix

**Comment by Musikid-Official** [comment]
I don't have the original link, but I do have the original image. I uploaded it to imgur and it worked, so it must have been something with the link.

I did try retaking the picture, too, and it worked. It must have had something to do with the original link, but Screencapper doesn't save those links as far as I can tell.

**Comment by Nettakrim** [comment]
yeah that sounds like the specific jpg was slightly corrupted/malformed or something and that just cascaded into a crash

**Comment by Nettakrim** [comment]
if you don't think you'll be able to reproduce it this issue can probably be marked as closed then reopened if it does happen again

**Comment by Nettakrim** [comment]
also you can set up windows to save a bit of clipboard history by pressing win+v, which can help with not accidentally losing stuff on the clipboard

**Comment by Musikid-Official** [comment]
I don't have the original link, but I do have the original image. I uploaded it to imgur and it worked, so it must have been something with the link.

I did try retaking the picture, too, and it worked. It must have had something to do with the original link, but Screencapper doesn't save those links as far as I can tell.

I found a different error, though: setting the pixel filter to 1 and the image smaller than 1x1 block it crashes. Prob just the rendering getting confused.
